ssf/libsaltcli.jinja

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# vim: ft=jinja
+
+{#- Get the relevant values from the `opts` dict #}
+{%- set opts_cli = opts.get('__cli', '') %}
+{%- set opts_masteropts_cli = opts | traverse('__master_opts__:__cli', '') %}
+{%- set opts_file_client = opts.get('file_client', '') %}
+
+{#- Determine the type of salt command being run #}
+{%- if opts_cli == 'salt-minion' %}
+{%-   set cli = 'minion' %}
+{%- elif opts_cli == 'salt-call' %}
+{%-   set cli = 'ssh' if opts_masteropts_cli == 'salt-ssh' else 'local' %}
+{%-   set cli = 'masterless' if opts_file_client == 'local' %}
```

Comment:
From my tests, `opts_file_client == 'local'` for `salt-ssh` as well as `salt-call --local`, i.e. for both types of `salt-call`.  Is that what you've got at your end?  If this is the case, then the implementation would need to be adjusted, because we're relying on being able to check `ssh` (and even `local`) in other formulas.

In fact, we could simply remove this line and import `opts_file_client` in the formula instead:

```sls
{%- from tplroot ~ "/libsaltcli.jinja" import opts_file_client with context %}
...
    {%- if opts_file_client != 'local' %}
    - runas: {{ ssf.user }}
    {%- endif %}
```

Reply:
> `opts_file_client == 'local'` for `salt-ssh`

Ok, I didn't realise that - I'll have to look at this again.